android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Intent.kt

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.ktx.android.content
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.content.pm.LabeledIntent
+import android.os.Build
+import android.os.Parcelable
+
+/**
+ * Modify the current intent to be used in an intent chooser excluding the current app.
+ *
+ * @param context Android context used for various system interactions.
+ * @param title Optional title that will be displayed in the chooser,
+ * only when the target action is not ACTION_SEND or ACTION_SEND_MULTIPLE.
+ *
+ * @return a new Intent object that you can hand to Context.startActivity() and related methods.
+ */
+fun Intent.createChooserExcludingCurrentApp(
+    context: Context,
+    title: CharSequence? = null,
+): Intent {
+    val chooserIntent: Intent
+    val resolveInfos = context.packageManager.queryIntentActivities(this, 0).toHashSet()
```

Comment:
I think we should keep `.queryIntentActivitiesCompat` to avoid us calling a deprecated method.